ui/raidboss/data/05-shb/raid/e8s.js

```diff
@@ -317,30 +317,201 @@
     },
     {
       id: 'E8S Refulgent Chain',
-      regex: Regexes.gainsEffect({ effect: 'Refulgent Chain' }),
-      regexDe: Regexes.gainsEffect({ effect: 'Lichtfessel' }),
-      regexFr: Regexes.gainsEffect({ effect: 'Chaînes de Lumière' }),
-      regexJa: Regexes.gainsEffect({ effect: '光の鎖' }),
-      condition: Conditions.targetIsYou(),
-      suppressSeconds: 1,
-      infoText: {
-        en: 'Chain on YOU',
-        de: 'Kette auf DIR',
-        fr: 'Chaine sur VOUS',
-        ko: '사슬 대상자',
-        cn: '连线',
+      regex: Regexes.tether({ id: '006E' }),
+      condition: function(data, matches) {
+        if (data.options.cactbote8sLightRampantStrat === undefined || data.options.cactbote8sLightRampantStrat === 'none')
+          return data.me == matches.source;
+
+        if (data.options.cactbote8sLightRampantStrat === 'sharingan') {
+          data.rampant = data.rampant || {};
+          data.rampant.chain = data.rampant.chain || {
+            chains: {},
+            total: 0,
+            meChain: false,
+            chainSwapTank: 'Stay',
+            chainSwapDps: 'Stay',
+          };
+          if (data.party.isTank(matches.source)) {
+            data.rampant.chain.chains.tank = matches;
+          } else {
+            if (data.party.isHealer(matches.source)) {
+              data.rampant.chain.chains.healer = matches;
+            } else {
+              if (data.rampant.chain.chains.dps1 === undefined)
+                data.rampant.chain.chains.dps1 = matches;
+              else
+                data.rampant.chain.chains.dps2 = matches;
+            }
+            if (data.party.isTank(matches.target))
+              data.rampant.chain.chains.tankReverse = matches;
+          }
+          ++data.rampant.chain.total;
+          data.rampant.chain.meChain = data.rampant.chain.meChain || data.me == matches.source;
+          if (data.rampant.chain.total===4) {
+            // First figure out which DPS is in which corner
+            // So we can ensure dps1 is southwest and dps2 is southeast
+            let meleePriority = ['MNK', 'DRG', 'NIN', 'SAM'];
```

Comment:
I like this a lot more than my current code, I'll do some testing. Not sure about the change of return text, since not everyone can run Cactbot (we have a tank and a DPS both on PS4, if they have to swap then there's no way for the person doing callouts to know that from Cactbot itself).

Reply:
This has been rewritten based on your proposal, fixed some bugs and adjusted the return conditions.